src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -91,6 +91,35 @@ pub fn update_lockfile(manifest_path: &Path,
                                                   Method::Everything,
                                                   Some(&previous_resolve),
                                                   Some(&to_avoid)));
+
+    // Summarize what is changing for the user.
+    let print_change = |status: &str, msg: String| {
+        opts.config.shell().status(status, msg)
+    };
+    for (removed, added) in compare_dependency_graphs(&previous_resolve, &resolve) {
+        if removed.len() == 1 && added.len() == 1 {
+            if removed[0].source_id().is_git() {
+                try!(print_change("Updating", format!("{} #{} -> #{}",
+                    removed[0].name(),
+                    &removed[0].source_id().precise().unwrap()[..8],
+                    &added[0].source_id().precise().unwrap()[..8])));
+            } else {
+                try!(print_change("Updating", format!("{} v{} -> v{}",
+                    removed[0].name(),
+                    removed[0].version(),
```

Comment:
These two lines can be replaced with `removed[0]` as it'll print the same thing for registry sources and otherwise print contextual information for things like path sources